### DISCUSSION TITLE
Recover the sources of a web page, gives me no data and a cors error again

### DISCUSSION THREAD

**Discussion by krxdow** [nuxt/nuxt#19421] [Questions] [answered]
hello everyone

I think to resolve my [issue](https://github.com/nuxt/nuxt/discussions/19276)  but not really

I try to get source code of web page on the server side, 
so i make this 

```javascript
  const {data: source} = await useFetch("https://www.jeuxdemots.org/rezo-dump.php?gotermsubmit=Chercher&gotermrel=crocodile&rel=&relin=norelin", {
    server:true,
    method: ['get','post'],
    mode: 'cors',
    headers: {
      'Access-Control-Allow-Origin': 'http://localhost:3000',
      'X-Requested-With': 'XMLHttpRequest',
    },
  })
  console.log("source:", toRaw(source.value))
  console.log(source)
}
```

I'm a little lost, with all possible options to set CORS (routeRule,plugins,middleware),
 with useFetch options, it seems I have no CORS error,
but  when i try to print with console.log method I have this on the console browser

` console.log(source)`
```javascript
Object { __v_isShallow: false, dep: undefined, __v_isRef: true, _rawValue: null, _value: null }
__v_isRef: true
__v_isShallow: false
_rawValue: null
_value: null
dep: undefined
```

and with  `console.log("source:", toRaw(source.value))`
in the browser console i have `source: null`
  
maybe the cors parameters are bad, but with these options I have no error in the browser console
if anyone has any leads i'm a taker
THANKS

**Comment by TheAlexLichter** [comment, marked as answer]
CORS headers have to be set by the page **you are requesting**, not in your fetch call. If the website linked above is yours, set it there. If not, you can only ask the author to add the headers as they have to be in **the response, not in your request**.

An easy way around the issue would be to proxy the URL via `routeRules` and call it locally (`useFetch('/my-site')`) because CORS only applies to other origins and not to your own domain.

**Reply by krxdow** [reply]
I already  tried several solutions via routeRules but i have CORS Missing Allow Origin error

````javascript
   routeRules: {
        '/**: { cors: true, headers: {
            'access-control-allow-methods': 'GET,POST', 
                'access-control-allow-origin:': '*',
                'access-control-allow-headers':'accept'
        }
        },
    },
````


maybe i have the wrong syntax

**Reply by TheAlexLichter** [reply]
I've updated the wording of my answer above in case it waswas not descriptive enough. You need to use the **proxy** option, not the **cors** option because **cors** wil make your site "available" to be queried by others.

**Reply by krxdow** [reply]
Thank you for your response 
I make this and it works
```javascript
    routeRules: {
               '/proxy/**'  : {
                   proxy: {to:'https://www.jeuxdemots.org/**'},
               },
    },
```

And on my vue page
````html
<template>
    <div>
        <Button @click="checkedWord" />
     </div>
</template>


<script setup>
 async function checkedWord() {
  const {data: response }=  await useFetch("http://127.0.0.1:3000/proxy/rezo-dump.php?gotermsubmit=Chercher&gotermrel=crocodile&rel=&relin=norelin")
console.log(toRaw(response.value))
}
</script>
````
